ucdlib-directory/src/editor/lib/blocks/ucdlib-directory-research-highlights/edit.js

```diff
@@ -0,0 +1,65 @@
+import { html, SelectUtils } from "@ucd-lib/brand-theme-editor/lib/utils";
+import { useBlockProps, InspectorControls } from '@wordpress/block-editor';
+import { PanelBody, TextControl, Button } from '@wordpress/components';
+import { Fragment, useState } from '@wordpress/element';
+import { useDispatch } from '@wordpress/data';
+
+export default () => {
+  const blockProps = useBlockProps();
+  const { editPost } = useDispatch('core/editor');
+
+  const rawMeta = SelectUtils.editedPostAttribute('meta') || {};
+
+  const expertId = rawMeta.aggie_experts_id || '';
+  const hideResearchHighlights = !!rawMeta.hide_research_highlights;
+  
+  const [query, setQuery] = useState(expertId);
```

Comment:
you do not need to manually control state.

you can call editPost directly from your `TextControl` `onChange`. The `Save Expert ID` button is superfluous.
https://github.com/UCDavisLibrary/ucdlib-wp-plugins/blob/main/ucdlib-directory/src/editor/lib/blocks/ucdlib-directory-bio/edit.js